out_writeapi_test.go

```diff
@@ -38,3 +56,150 @@ func TestFLBPluginRegister(t *testing.T) {
 	assert.Equal(t, "writeapi", currplugin.name)
 
 }
+
+// this is a struct keeping track of whether the correct options are sent in NewManagedStream
+type OptionChecks struct {
+	configProjectID        bool
+	configDatasetID        bool
+	configTableID          bool
+	configMaxChunkSize     bool
+	configMaxQueueSize     bool
+	configMaxQueueRequests bool
+	calledGetClient        int
+	calledNewManagedStream int
+	calledGetWriteStream   int
+	calledSetContext       int
+	numInputs              bool
+	mapSizeIncremented     bool
+}
+
+type MockManagedWriterClient struct {
+	client                      *managedwriter.Client
+	NewManagedStreamFunc        func(ctx context.Context, opts ...managedwriter.WriterOption) (*managedwriter.ManagedStream, error)
+	GetWriteStreamFunc          func(ctx context.Context, req *storagepb.GetWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error)
+	CloseFunc                   func() error
+	BatchCommitWriteStreamsFunc func(ctx context.Context, req *storagepb.BatchCommitWriteStreamsRequest, opts ...gax.CallOption) (*storagepb.BatchCommitWriteStreamsResponse, error)
+	CreateWriteStreamFunc       func(ctx context.Context, req *storagepb.CreateWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error)
+}
+
+func (m *MockManagedWriterClient) NewManagedStream(ctx context.Context, opts ...managedwriter.WriterOption) (*managedwriter.ManagedStream, error) {
+	return m.NewManagedStreamFunc(ctx, opts...)
+}
+
+func (m *MockManagedWriterClient) GetWriteStream(ctx context.Context, req *storagepb.GetWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+	return m.GetWriteStreamFunc(ctx, req, opts...)
+}
+
+func (m *MockManagedWriterClient) Close() error {
+	return m.client.Close()
+}
+
+func (m *MockManagedWriterClient) BatchCommitWriteStreams(ctx context.Context, req *storagepb.BatchCommitWriteStreamsRequest, opts ...gax.CallOption) (*storagepb.BatchCommitWriteStreamsResponse, error) {
+	return m.client.BatchCommitWriteStreams(ctx, req, opts...)
+}
+
+func (m *MockManagedWriterClient) CreateWriteStream(ctx context.Context, req *storagepb.CreateWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+	return m.client.CreateWriteStream(ctx, req, opts...)
+}
+
+// TestFLBPluginInit tests the FLBPluginInit function
+func TestFLBPluginInit(t *testing.T) {
+	//variable keeping track of what order functions are called in
+	order := 0
```

Comment:
Checking for the order in which the functions are called seems excessive and verging on implementation-dependent. We generally don't do that. Verifying that each function is called the expected number of times is sufficient.

Reply:
Done. I also deleted the log file so you don't see anything. I'm not sure how else to omit the changes.